cgrok/grok.go

```diff
@@ -46,10 +49,23 @@ func New() *Grok {
 	if grok.g == nil {
 		return nil
 	}
-
+	grok.stringCache = make(map[uintptr]string)
 	return grok
 }
 
+/* Only look up group names, which live as long as the grok and are frequently reused. 
+   With other strings there's a risk of getting the wrong result if memory has been freed and reused. */
+func (grok *Grok) gostringn(str *C.char, len C.int) string {
```

Comment:
This is a nice little caching strategy. I think the answer is no, but do we need to worry about running `free` on any of these strings?